subgraphs/bridgeworld/src/mappings/advanced-questing.ts

```diff
@@ -0,0 +1,194 @@
+  );
+
+  quest.requestId = params._requestId;
+  quest.token = getAddressId(LEGION_ADDRESS, params._startQuestParams.legionId);
+  quest.user = params._owner.toHexString();
+  quest.status = "Idle";
+  quest.zoneName = params._startQuestParams.zoneName;
+  quest.part = params._startQuestParams.advanceToPart;
+
+  const treasures: string[] = [];
+
+  for (let i = 0; i < params._startQuestParams.treasureIds.length; i++) {
+    const tokenId = params._startQuestParams.treasureIds[i].toHex();
+
+    const treasure = new TokenQuantity(
+      `${quest.id}-${quest.requestId}-${tokenId}`
+    );
+    treasure.token = tokenId;
+    treasure.quantity = params._startQuestParams.treasureAmounts[i].toI32();
+
+    treasure.save();
+
+    treasures.push(treasure.id);
+  }
+
+  quest.treasures = treasures;
+
+  updateQuestEndTimeAndStasis(quest, params._startQuestParams.legionId);
+
+  quest.save();
+}
+
+export function handleAdvancedQuestContinued(
+  event: AdvancedQuestContinued
+): void {
+  const params = event.params;
+
+  const id = getAddressId(event.address, params._legionId);
+  const quest = AdvancedQuest.load(id);
+
+  if (!quest) {
+    log.error("[advanced-quest-started] Unknown quest: {}", [id]);
+
+    return;
+  }
+
+  quest.part = params._toPart;
+  updateQuestEndTimeAndStasis(quest, params._legionId);
+
+  quest.save();
+}
+
+export function handleTreasureTriadPlayed(event: TreasureTriadPlayed): void {
+  const params = event.params;
+
+  const id = getAddressId(event.address, params._legionId);
+  const result = new TreasureTriadResult(id);
+  const quest = AdvancedQuest.load(id);
+
+  if (!quest) {
+    log.error("[advanced-quest-treasure-triad-played] Unknown quest: {}", [id]);
+
+    return;
+  }
+
+  result.advancedQuest = quest.id;
+  result.playerWon = params._playerWon;
+  result.numberOfCardsFlipped = params._numberOfCardsFlipped;
+  result.numberOfCorruptedCardsRemaining =
+    params._numberOfCorruptedCardsRemaining;
+
+  quest.treasureTriadResult = result.id;
+
+  result.save();
+  quest.save();
+}
+
+export function handleAdvancedQuestEnded(event: AdvancedQuestEnded): void {
+  const params = event.params;
+
+  const id = getAddressId(event.address, params._legionId);
+  const quest = AdvancedQuest.load(id);
+  const treasureTriadResult = TreasureTriadResult.load(id);
+
+  if (!quest) {
+    log.error("[advanced-quest-ended] Unknown quest: {}", [id]);
+
+    return;
+  }
+
+  quest.id = `${quest.id}-${quest.requestId}`;
+  quest.status = "Finished";
+
+  store.remove("AdvancedQuest", id);
+
+  for (let i = 0; i < params._rewards.length; i++) {
+    const rewardId = `${quest.id}-${i}`;
+    const reward = new AdvancedQuestReward(rewardId);
+
+    reward.advancedQuest = quest.id;
+
+    if (params._rewards[i].consumableId.toI32() !== 0) {
+      const consumable = new TokenQuantity(
+        `${rewardId}-${params._rewards[i].consumableId.toHex()}`
+      );
+      consumable.token = params._rewards[i].consumableId.toHex();
```

Comment:
This needs to point to the `id` of the token, not just the `tokenId`.

```suggestion
      consumable.token = getAddressId(CONSUMABLE_ADDRESS, params._rewards[i].consumableId);
```